.semaphore/semaphore.yml

```diff
@@ -4,15 +4,47 @@ agent:
   machine:
     type: e1-standard-4
     os_image: ubuntu1804
+global_job_config:
+  prologue:
+    commands:
+      - checkout
+      # Semaphore is doing shallow clone on a commit without tags.
+      # unshallow it for GIT_VERSION:=$(shell git describe --tags --dirty --always) @ Makefile.common
+      - git fetch --unshallow
```

Comment:
perhaps consider using [semaphore cache](https://docs.semaphoreci.com/ci-cd-environment/toolbox-reference/#cache) instead of having each block checking out the same code

Reply:
I think I should merge this as-is to unblock CI, but that's an interesting feature - I will try to put up a follow-on PR that incorporates that. Would be nice to avoid downloading and building the code multiple times. 

maybe we could do a 
- Download, build, cache step
- Other steps un-cache the code as well as build artifacts, so they don't need even need to re-build?